### DISCUSSION TITLE
option to override stat -x behaviour on mac os x

### DISCUSSION THREAD

**Discussion by dvic** [jarun/nnn#1756] [Ideas]
I'm using coreutils (installed with brew) on my mac, so for me `stat -x` gives an error. Should we have a compiler flag that overrides the behaviour of adding the `-x` option on mac?

**Comment by jarun** [comment]
We never came across this from any other user on macOS. The code also uses the `-x` option for macOS.

```
4617 #ifdef FILE_MIME_OPTS
4618                 ("file " FILE_MIME_OPTS),
4619 #endif
4620                 "file -b",
4621 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__)
4622                 "stat -x",                                                                                         
4623 #else
4624                 "stat",
4625 #endif
```

Till we hear from more users we can't go ahead and change it. Feel free to patch the code.

**Comment by dvic** [comment]
Yeah I started using the gnu version of the tools because I was annoyed encountering each time the difference between mac and linux but now it bites back 😅 let me know if/when there’s interest for a PR that lets you override this with an env variabel.
